File: app/src/main/java/com/example/product1/TovarListFragment.java
package com.example.product1;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;


public class TovarListFragment extends Fragment {
    String[] tovar = { "Понеділок", "Вівторок", "Середа", "Четвер", "Пятниця"};

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_tovar_list, container, false);
        ListView tovarList = (ListView) view.findViewById(R.id.tovarList);
        final Intent intent = new Intent(view.getContext(), TovarDetails.class);

        ArrayAdapter<String> adapter = new ArrayAdapter(view.getContext(),
                android.R.layout.simple_list_item_1, tovar);

        tovarList.setAdapter(adapter);

        tovarList.setOnItemClickListener(new AdapterView.OnItemClickListener(){
            @Override
            public void onItemClick(AdapterView<?> parent, View v, int position, long id)
            {
                startActivity(intent);
            }
        });

        return  view;
    }
}

File: app/src/main/java/com/example/product1/tovarslist/TovarListFragment.kt
package com.example.product1.tovarslist;

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.example.product1.R
import com.example.product1.tovardetails.TovarDetails


class TovarListFragment : Fragment() {

    private lateinit var viewModel: TovarListViewModel
    private  lateinit var viewModelFactory: TovarListViewModelFactory

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        var view : View = inflater?.inflate(R.layout.fragment_tovar_list, container, false);

        viewModelFactory = TovarListViewModelFactory()
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(TovarListViewModel::class.java)


        var contactsList:ListView =  view?.findViewById(R.id.tovarList);

        var adapter = ArrayAdapter<String>(view.getContext(),
                android.R.layout.simple_list_item_1, viewModel.tovar);

        contactsList.setAdapter(adapter);
        contactsList.setOnItemClickListener { parent, view, position, id ->
            activity?.supportFragmentManager?.beginTransaction()?.replace(R.id.activity_main, TovarDetails())?.commit()
            //activity?.supportFragmentManager?.beginTransaction()?.remove(ContactsListFragment())?.commit()
         }

        return  view;
    }
}